src/autoscan.cc

```diff
@@ -174,13 +168,12 @@ int AutoscanList::removeByObjectID(int objectID)
 {
     AutoLock lock(mutex);
 
-    for (size_t i = 0; i < list.size(); i++) {
-        if (objectID == list[i]->getObjectID()) {
-            auto dir = list[i];
-            dir->setScanID(INVALID_SCAN_ID);
-            list.erase(list.begin() + i);
-            return i;
-        }
+    auto it = std::find_if(list.begin(), list.end(), [=](const auto& item) { return objectID == item->getObjectID(); });
+    if (it != list.end()) {
```

Comment:
I'd like (and some agree) it's better to fail fast: no item, return an error rather than doing work in the `if` clause.

Another thing: do you really need to copy `dir`?

Reply:
src/autoscan.cc:174:14: error: no member named 'setScanID' in 'std::shared_ptr<AutoscanDirectory>'
        *it->setScanID(INVALID_SCAN_ID);

Failing fast is a good idea.

Reply:
`(*it)->...` i think it's operator precedence trick.